src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -50,23 +55,39 @@ public static void setHeader(final SAMFileHeader header) {
      * @param outputFile path to the output bam.
      * @param rddReads reads to write.
      * @param header the header to put at the top of the files
-     * @param singleFile should the output be a single file or sharded, i.e., outputfile/part-r-00000, outputfile/part-r-00001...
+     * @param format should the output be a single file, sharded, ADAM, etc.
      */
     public static void writeReads(
             final JavaSparkContext ctx, final String outputFile, final JavaRDD<GATKRead> rddReads,
-            final SAMFileHeader header, boolean singleFile) throws IOException {
-        if (singleFile) {
+            final SAMFileHeader header, ReadsWriteFormat format) throws IOException {
+        if (format.equals(ReadsWriteFormat.SINGLE)) {
             writeReadsSingle(ctx, outputFile, rddReads, header);
-        } else {
+        } else if (format.equals(ReadsWriteFormat.SHARDED)) {
             writeReadsSharded(ctx, outputFile, rddReads, header);
-
+        } else if (format.equals(ReadsWriteFormat.ADAM)) {
+            writeReadsADAM(ctx, outputFile, rddReads, header);
         }
+    }
 
+    private static void writeReadsADAM(
+            final JavaSparkContext ctx, final String outputFile, final JavaRDD<GATKRead> rddGATKReads,
+            final SAMFileHeader header) throws IOException {
+        JavaPairRDD<Void, AlignmentRecord> rddAlignmentRecords =
+                rddGATKReads.map(new GATKReadToBDGAlignmentRecordConverter(header)).mapToPair(
+                        alignmentRecord -> new Tuple2<Void, AlignmentRecord>(null, alignmentRecord));
+        // instantiating a Job is necessary here in order to set the Hadoop Configuration...
+        Job job = Job.getInstance(ctx.hadoopConfiguration());
+        // ...here, which sets a config property that the AvroParquetOutputFormat needs when writing data. Specifically,
+        // we are writing the Avro schema to the Configuration as a JSON string. The AvroParquetOutputFormat class knows
+        // how to translate objects in the Avro data model to the Parquet primitives that get written.
+        AvroParquetOutputFormat.setSchema(job, AlignmentRecord.getClassSchema());
```

Comment:
Off topic:
Just curious, why doesn't AvroParquetOutputFormat.setSchema take a configuration class instead of `Job`? It seems like that's what it modifies.